internal/testrunner/testrunner.go

```diff
@@ -284,13 +287,49 @@ func RegisterRunner(runner TestRunner) {
 	runners[runner.Type()] = runner
 }
 
-// Run method delegates execution to the registered test runner, based on the test type.
-func Run(ctx context.Context, testType TestType, options TestOptions) ([]TestResult, error) {
-	runner, defined := runners[testType]
+type runnerLauncher struct {
+	logger *slog.Logger
+}
+
+func NewRunnerLauncher(options ...RunnerLauncherOption) *runnerLauncher {
+	r := runnerLauncher{logger: logger.Logger}
+	for _, opt := range options {
+		opt(&r)
+	}
+
+	return &r
+}
+
+type RunnerLauncherOption func(r *runnerLauncher)
+
+func WithLogger(logger *slog.Logger) RunnerLauncherOption {
+	return func(r *runnerLauncher) {
+		r.logger = logger
+	}
+}
```

Comment:
Another option would be to set a new method in the `TestRunner` interface (e.g. `SetLogger()`), so it can be updated the logger of each runner dynamically.

Reply:
Updated with this option in https://github.com/elastic/elastic-package/pull/1876/commits/6cb19a0dfff335a04e46ab3d387f2ecfc781754b